internal/otlptext/databuffer_test.go

```diff
@@ -24,27 +24,27 @@ import (
 
 func TestNestedArraySerializesCorrectly(t *testing.T) {
```

Comment:
I'm OK with the changes from this PR, but I feel like it could also have a "proof" that this is backwards compatible. Like, by keeping this test using the old functions, changing it only when the old ones are removed. Or duplicating this test, with the one using the old functions being removed when the functions are removed.

Reply:
Copied the test using old funcs with `Deprecated` suffix. I did the same for this test already https://github.com/open-telemetry/opentelemetry-collector/pull/4909/files#diff-5e341afb2ee55aae2edd3b7b3b3d714dc5d2233ec814131a720f74b02de8754fR28